### PR TITLE
[11.x] Add Coveralls CI workflow for test coverage reporting

### PR DESCRIPTION
**Pull request by joelbladt**
## Description
This Pull Request introduces a new GitHub Actions workflow to integrate Coveralls for automated test coverage reporting. The workflow is added under `.github/workflows/coveralls.yml`. It is designed to provide detailed code coverage metrics for pull requests and the main branches.

## Benefit to End Users
The inclusion of Coveralls enhances the development workflow by providing:

* Transparent Code Coverage Metrics: Developers and contributors gain insights into the impact of their changes on the overall code coverage.
* Improved Code Quality: Encourages maintaining and improving code coverage across the framework.
* Better Collaboration: Simplifies identifying areas of insufficient test coverage during code reviews.

These benefits ensure a more robust and reliable Laravel framework for developers and end users alike.

## Implementation Details
1. **Workflow Name:** `coveralls`
2. **Triggers:** The workflow runs on:
* Push events to the `master` branch and branches matching the `*.x` pattern.
* All pull requests.
3. **Jobs Configuration:**
* Runs on `ubuntu-24.04` with a matrix of PHP versions (`8.2`, `8.3`, `8.4`) and PHPUnit versions (`10.5.12`, `11.3.2`).
* Executes tests with `phpunit` and generates a Clover coverage report.
* Uploads the coverage data to Coveralls using the `php-coveralls` library.

The workflow does not interfere with existing configurations and complements the current CI/CD processes.

## Why It Doesn't Break Existing Features
* The workflow operates independently of existing testing and deployment pipelines.
* No changes were made to the core functionality or APIs of the Laravel framework.
* The coveralls.yml file is isolated and only concerns reporting test coverage metrics.

## How It Makes Building Web Applications Easier
By integrating Coveralls, contributors can:

* Ensure that new code is properly tested and covered.
* Quickly identify areas in the framework that need more rigorous testing.
* Enhance trust and confidence in the stability of the framework with every new change.

## Tests
No additional feature tests are required for this change since it pertains to CI workflow. However:

* The workflow has been tested locally and on other repositories to ensure compatibility with the defined configurations and tools (PHP, PHPUnit, Coveralls).
* Verified that coverage reports are generated and uploaded successfully to Coveralls.

### CODE REVIEW

File: .github/workflows/coveralls.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+jobs:
+  types:
+    runs-on: ubuntu-24.04
+
+    strategy:
+      fail-fast: true
+      matrix:
+        php: [ 8.2, 8.3, 8.4 ]
+        phpunit: [ '10.5.12', '11.3.2' ]
```
Comment (crynobone) [inline .github/workflows/coveralls.yml:18]: It should just run on a single PHP version and the latest PHPUnit.

File: .github/workflows/databases-nightly.yml [hunk 2]
```diff
@@ -31,9 +31,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.3
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, xml, xdebug, :php-psr
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/databases-nightly.yml:36]:
```suggestion
          coverage: none
```

File: .github/workflows/databases-nightly.yml [hunk 3]
```diff
@@ -77,9 +77,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.3
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/databases-nightly.yml:80]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
```

File: .github/workflows/databases-nightly.yml [hunk 4]
```diff
@@ -77,9 +77,9 @@ jobs:
...
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/databases-nightly.yml:82]:
```suggestion
          coverage: none
```

File: .github/workflows/databases-nightly.yml [hunk 5]
```diff
@@ -31,9 +31,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.3
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/databases-nightly.yml:34]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
```

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -133,9 +133,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ matrix.php }}
-          extensions: dom, curl, libxml, mbstring, zip, pdo, sqlite, pdo_sqlite, gd, pdo_mysql, fileinfo, ftp, redis, memcached, gmp, intl, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pdo, sqlite, pdo_sqlite, gd, pdo_mysql, fileinfo, ftp, redis, memcached, gmp, intl, xml, xdebug, :php-psr
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/tests.yml:138]:
```suggestion
          coverage: none
```

File: .github/workflows/tests.yml [hunk 7]
```diff
@@ -133,9 +133,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ matrix.php }}
-          extensions: dom, curl, libxml, mbstring, zip, pdo, sqlite, pdo_sqlite, gd, pdo_mysql, fileinfo, ftp, redis, memcached, gmp, intl, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pdo, sqlite, pdo_sqlite, gd, pdo_mysql, fileinfo, ftp, redis, memcached, gmp, intl, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/tests.yml:136]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pdo, sqlite, pdo_sqlite, gd, pdo_mysql, fileinfo, ftp, redis, memcached, gmp, intl, :php-psr
```

File: .github/workflows/tests.yml [hunk 8]
```diff
@@ -53,10 +53,10 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ matrix.php }}
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, sqlite, pdo_sqlite, gd, redis, igbinary, msgpack, memcached, gmp, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, sqlite, pdo_sqlite, gd, redis, igbinary, msgpack, memcached, gmp, xml, xdebug, :php-psr
           ini-values: error_reporting=E_ALL
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/tests.yml:59]:
```suggestion
          coverage: none
```

File: .github/workflows/tests.yml [hunk 9]
```diff
@@ -53,10 +53,10 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ matrix.php }}
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, sqlite, pdo_sqlite, gd, redis, igbinary, msgpack, memcached, gmp, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, sqlite, pdo_sqlite, gd, redis, igbinary, msgpack, memcached, gmp, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/tests.yml:56]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, sqlite, pdo_sqlite, gd, redis, igbinary, msgpack, memcached, gmp, :php-psr
```

File: .github/workflows/queues.yml [hunk 10]
```diff
@@ -147,9 +147,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.2
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, xml, xdebug, :php-psr
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/queues.yml:152]:
```suggestion
          coverage: none
```

File: .github/workflows/queues.yml [hunk 11]
```diff
@@ -147,9 +147,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.2
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/queues.yml:150]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
```

File: .github/workflows/queues.yml [hunk 12]
```diff
@@ -107,9 +107,9 @@ jobs:
...
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/queues.yml:112]:
```suggestion
          coverage: none
```

File: .github/workflows/queues.yml [hunk 13]
```diff
@@ -107,9 +107,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.2
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/queues.yml:110]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
```

File: .github/workflows/queues.yml [hunk 14]
```diff
@@ -59,9 +59,9 @@ jobs:
...
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/queues.yml:64]:
```suggestion
          coverage: none
```

File: .github/workflows/queues.yml [hunk 15]
```diff
@@ -59,9 +59,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.2
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/queues.yml:62]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
```

File: .github/workflows/queues.yml [hunk 16]
```diff
@@ -24,9 +24,9 @@ jobs:
...
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/queues.yml:29]:
```suggestion
          coverage: none
```

File: .github/workflows/queues.yml [hunk 17]
```diff
@@ -24,9 +24,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.2
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/queues.yml:27]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
```

File: .github/workflows/databases.yml [hunk 18]
```diff
@@ -358,9 +358,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.3
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, sqlsrv, pdo, pdo_sqlsrv, odbc, pdo_odbc, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, sqlsrv, pdo, pdo_sqlsrv, odbc, pdo_odbc, xml, xdebug, :php-psr
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/databases.yml:363]:
```suggestion
          coverage: none
```

File: .github/workflows/databases.yml [hunk 19]
```diff
@@ -358,9 +358,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.3
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, sqlsrv, pdo, pdo_sqlsrv, odbc, pdo_odbc, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, sqlsrv, pdo, pdo_sqlsrv, odbc, pdo_odbc, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/databases.yml:361]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pcntl, sqlsrv, pdo, pdo_sqlsrv, odbc, pdo_odbc, :php-psr
```

File: .github/workflows/databases.yml [hunk 20]
```diff
@@ -319,9 +319,9 @@ jobs:
...
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/databases.yml:324]:
```suggestion
          coverage: none
```

File: .github/workflows/databases.yml [hunk 21]
```diff
@@ -319,9 +319,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.3
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, sqlsrv, pdo, pdo_sqlsrv, odbc, pdo_odbc, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, sqlsrv, pdo, pdo_sqlsrv, odbc, pdo_odbc, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/databases.yml:322]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pcntl, sqlsrv, pdo, pdo_sqlsrv, odbc, pdo_odbc, :php-psr
```

File: .github/workflows/databases.yml [hunk 22]
```diff
@@ -271,9 +271,9 @@ jobs:
...
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/databases.yml:276]:
```suggestion
          coverage: none
```

File: .github/workflows/databases.yml [hunk 23]
```diff
@@ -271,9 +271,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.3
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, sqlsrv, pdo, pdo_sqlsrv, odbc, pdo_odbc, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, sqlsrv, pdo, pdo_sqlsrv, odbc, pdo_odbc, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/databases.yml:274]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pcntl, sqlsrv, pdo, pdo_sqlsrv, odbc, pdo_odbc, :php-psr
```

File: .github/workflows/databases.yml [hunk 24]
```diff
@@ -224,9 +224,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.3
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_pgsql, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_pgsql, xml, xdebug, :php-psr
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/databases.yml:229]:
```suggestion
          coverage: none
```

File: .github/workflows/databases.yml [hunk 25]
```diff
@@ -224,9 +224,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.3
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_pgsql, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_pgsql, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/databases.yml:227]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_pgsql, :php-psr
```

File: .github/workflows/databases.yml [hunk 26]
```diff
@@ -175,9 +175,9 @@ jobs:
...
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/databases.yml:180]:
```suggestion
          coverage: none
```

File: .github/workflows/databases.yml [hunk 27]
```diff
@@ -175,9 +175,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.3
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_pgsql, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_pgsql, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/databases.yml:178]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_pgsql, :php-psr
```

File: .github/workflows/databases.yml [hunk 28]
```diff
@@ -128,9 +128,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.3
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, xml, xdebug, :php-psr
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/databases.yml:133]:
```suggestion
          coverage: none
```

File: .github/workflows/databases.yml [hunk 29]
```diff
@@ -128,9 +128,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.3
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/databases.yml:131]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
```

File: .github/workflows/databases.yml [hunk 30]
```diff
@@ -82,9 +82,9 @@ jobs:
...
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/databases.yml:87]:
```suggestion
          coverage: none
```

File: .github/workflows/databases.yml [hunk 31]
```diff
@@ -82,9 +82,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.3
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/databases.yml:85]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
```

File: .github/workflows/databases.yml [hunk 32]
```diff
@@ -35,9 +35,9 @@ jobs:
...
           tools: composer:v2
-          coverage: none
+          coverage: xdebug
```
Comment (crynobone) [inline .github/workflows/databases.yml:40]:
```suggestion
          coverage: none
```

File: .github/workflows/databases.yml [hunk 33]
```diff
@@ -35,9 +35,9 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: 8.3
-          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
+          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, xml, xdebug, :php-psr
```
Comment (crynobone) [inline .github/workflows/databases.yml:38]:
```suggestion
          extensions: dom, curl, libxml, mbstring, zip, pcntl, pdo, pdo_mysql, :php-psr
```

**Review by crynobone — CHANGES_REQUESTED** [review]

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If applicable, please consider releasing your code as a package so that the community can still take advantage of your contributions!

**Pull request closed without merging**